index.js

```diff
@@ -265,7 +263,7 @@ function rateLimitRequestHandler (params, pluginComponent) {
 
     if (code === 403) {
       respCtx.ban = true
-      params.onBanReach(req, key)
+      if (typeof params.onBanReach === 'function') params.onBanReach(req, key)
```

Comment:
Uff. I dont think this is nice. the noop can be optimized by v8, but this is basically a string compare, and is probably slower.

Reply:
I see, I though about function inlining even though that parameter is dynamic, which made me unsure

What about these then? Wouldn't replacing them with empty functions too better then?
```js
      if (typeof params.onExceeding === 'function') {
        params.onExceeding(req, key)
      }
      return
    }
    if (typeof params.onExceeded === 'function') {
      params.onExceeded(req, key)
    }
```

Reply:
Let me test actually

Reply:
Anyway if Fastify is all for speed, like you said, it shouldn't have typeof checks in request handlers

The types are there for a reason